src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
-    public static void main(String[] args) {
+    //List of Special User Commands
+    public static final String EXIT_STRING = "bye";
+    public static final String LIST_COMMAND = "list";
+    public static final String DONE_COMMAND = "done";
+    public static final String TODO_COMMAND = "todo";
+    public static final String DEADLINE_COMMAND = "deadline";
+    public static final String EVENT_COMMAND = "event";
+
+    //Other constants
+    public static final int NUM_OF_TASKS = 100;
+    public static final String LINE = "____________________________________________________________";
+    public static final String EVENT_KEYWORD = " /at";
+    public static final String DEADLINE_KEYWORD = " /by";
+    public static final String GOODBYE_MESSAGE = " Bye. Hope to see you again soon!";
+    public static final String EXCEPTION = " Sorry, your input is invalid! Please enter a valid input :)";
+
+    public static void printList(Task[] taskList) {
+        if (taskList.length == 0) {
+            System.out.println(LINE);
+            System.out.println(" No Tasks here yet. Go include some tasks!");
+            System.out.println(LINE);
+            System.out.println();
+            return;
+        }
+
+        System.out.println(LINE);
+        System.out.println(" Here are the tasks in your list:");
+        for (int i = 0; i < taskList.length; i++) {
+            System.out.println(" " + (i + 1) + ". " + taskList[i].toString());
+        }
+        System.out.println(LINE);
+        System.out.println();
+    }
+
+    public static void printDukeGreet() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        //Greeting the User
+        System.out.println(LINE);
+        System.out.println(" Hello! I'm Duke");
+        System.out.println(" What can I do for you?");
+        System.out.println(LINE);
+    }
+
+    public static boolean isNumeric(String strNum) {
+        if (strNum == null) {
+            return false;
+        }
+        try {
+            int num = Integer.parseInt(strNum);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+        return true;
+    }
+
+    public static void returnException() {
+        System.out.println(LINE);
+        System.out.println(EXCEPTION);
+        System.out.println(LINE);
+        System.out.println();
```

Comment:
Can use the System line Separator instead of printing a empty string on a new line, as this can cause some confusion to readers

Reply:
Thank you for the insightful feedback, will address them in my next update